code/modules/projectiles/projectile/beams.dm

```diff
@@ -184,3 +184,150 @@
 		var/mob/living/carbon/M = target
 		M.visible_message("<span class='danger'>[M] explodes into a shower of gibs!</span>")
 		M.gib()
+
+/obj/item/projectile/beam/heavy
+	name = "\improper heavy laser beam"
+	icon_state = "heavylaser"
+	damage = 35
+	irradiate = 30
+	armour_penetration = 30
+	dismemberment = 10
+	light_color = LIGHT_COLOR_RED
+	tracer_type = /obj/effect/projectile/tracer/heavy_laser
+	muzzle_type = /obj/effect/projectile/muzzle/heavy_laser
+	impact_type = /obj/effect/projectile/impact/heavy_laser
+	pass_flags = PASSTABLE
+
+obj/item/projectile/beam/heavy/on_hit(atom/target, blocked = FALSE)
+	. = ..()
+	if (!QDELETED(target) && (isturf(target) || istype(target, /obj/structure/) ||  istype(target, /obj/machinery/)))
+		target.ex_act(EXPLODE_LIGHT)
+
+/obj/item/projectile/beam/shock
+	name = "\improper charged beam"
+	icon_state = "spark"
+	damage = 10
+	jitter = 10
+	stutter = 10
+	hitsound = 'sound/weapons/wave.ogg'
+	light_color = LIGHT_COLOR_CYAN
+	tracer_type = /obj/effect/projectile/tracer/stun
+	muzzle_type = /obj/effect/projectile/tracer/stun
+	impact_type = /obj/effect/projectile/impact/stun
+
+/obj/item/projectile/beam/shock/on_hit(atom/target)
+	if(iscarbon(target))
+		var/mob/living/carbon/C = target
+		C.electrocute_act(10, src, 1, FALSE, FALSE, FALSE, FALSE, FALSE)
+		C.confused += 6
+		if(prob(20))
+			C.dropItemToGround(C.get_active_held_item())
+
+/obj/item/projectile/beam/bitcoin
+	name = "\improper bitcoin stealing beam"
+	icon_state = "bitcoin"
+	damage = 0
+	nodamage = TRUE
+	hitsound = 'sound/weapons/cash.ogg'
+	impact_effect_type = /obj/effect/temp_visual/impact_effect/green_laser
+	light_color = LIGHT_COLOR_GREEN
+	tracer_type = /obj/effect/projectile/tracer/xray
+	muzzle_type = /obj/effect/projectile/muzzle/xray
+	impact_type = /obj/effect/projectile/impact/xray
+
+/obj/item/projectile/beam/bitcoin/on_hit(atom/target)
+	. = ..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		var/obj/item/card/id/C = H.get_idcard(TRUE)
+		if(C && C.registered_account && !(C.registered_account.account_balance < 50))
+			C.registered_account.adjust_money(-50)
+			to_chat(H, "<span class='danger'>Your wallet feels lighter!</span>")
+			if(ishuman(firer))
+				var/mob/living/carbon/human/FH = firer
+				var/obj/item/card/id/FC = FH.get_idcard(TRUE)
+				if(FC && FC.registered_account)
+					FC.registered_account.adjust_money(50)
+					to_chat(FH, "<span class='notice'>Your wallet has received 50 credits.</span>")
+			else
+				var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_SEC)
+				if(D)
+					D.adjust_money(50)
+		else
+			H.adjustStaminaLoss(40)
+			to_chat(H, "<span class='danger'>You feel your energy itself being turned into cryptocurrency!</span>")

```

Comment:
This message isn't even true anymore and making it true seems like a very easily abusable feature

Reply:
40 stamloss is too high under the new system, needs adjustment